### DISCUSSION TITLE
Failed to publish to App Store Connect

### DISCUSSION THREAD

**Discussion by samfreeman210** [codemagic-ci-cd/codemagic-docs#1924] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

Imported flutterflow project from git hub repository to codemagic
Codemagic deploys to IOS app store connect 

### Expected results

Code magic deploys the git hub repository to IOS 
Happiness 

### Actual results

tool-version": "6.315.14315",
    "tool-path": "/Applications/Xcode-14.3.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/Frameworks/AppStoreService.framework",
    "os-version": "13.3.1",
    "product-errors": [
        {
            "message": "The provided entity includes an attribute with a value that has already been used",
            "userInfo": {
                "NSUnderlyingError": "Error Domain=IrisAPI Code=-19241 \"The provided entity includes an attribute with a value that has already been used\" UserInfo={status=409, detail=The bundle version must be higher than the previously uploaded version., source={\n    pointer = \"/data/attributes/cfBundleVersion\";\n}, id=2b462983-86e8-405f-be6c-bde93d8bf324, code=ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE, title=The provided entity includes an attribute with a value that has already been used, meta={\n    previousBundleVersion = 1;\n}, NSLocalizedDescription=The provided entity includes an attribute with a value that has already been used, NSLocalizedFailureReason=The bundle version must be higher than the previously uploaded version.}",
                "NSLocalizedDescription": "The provided entity includes an attribute with a value that has already been used",
                "previousBundleVersion": "1",
                "iris-code": "ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE",
                "NSLocalizedFailureReason": "The bundle version must be higher than the previously uploaded version: \u20181\u2019. (ID: 2b462983-86e8-405f-be6c-bde93d8bf324)"
            },
            "code": -19232
        }
    ]
}
Failed to upload archive at "/Users/builder/clone/build/ios/ipa/Lace.ipa"
Failed to publish /Users/builder/clone/build/ios/ipa/Lace.ipa

Failed to publish Lace.ipa to App Store Connect.

### Build id (optional)

64c394b6a71cfc8572b6610b

**Comment by himesh-cm** [comment]
Hi @samfreeman210, i have already replied to you on the intercom. Let's continue the conversation there.

**Comment by himesh-cm** [comment]
The current error is related to the bundle version, as per the below error message.
The bundle version must be higher than the previously uploaded version.

One way to fix it by specifying the version to your build ipa command as below
`--build-name=1.0.$PROJECT_BUILD_NUMBER --build-number=$PROJECT_BUILD_NUMBER`

For more reference on Flutter build versioning, please check the doc [here](https://docs.codemagic.io/knowledge-codemagic/build-versioning/).

**Reply by omar-chaabouni** [reply]
Yes thank you !

**Comment by himesh-cm** [comment]
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!

**Comment by francescorinaldi** [comment]
I got the same error, but I'm using a higher bundle version than the previously updated version,

```
VERSION_NAME=1.0
BUILD_NUMBER=$(($(app-store-connect get-latest-testflight-build-number "$APP_ID") + 1))   
echo $BUILD_NUMBER

flutter build ipa --release \
  --build-name=$VERSION_NAME \
  --build-number=$BUILD_NUMBER \
  --export-options-plist=/Users/builder/export_options.plist
```

This is the echo output:

```
Found build number 16 from TestFlight version 1.0
17
```

and this is the error:

```
== Gathering artifacts ==

== Publishing artifacts ==

Publishing artifact xxx.ipa
Publishing xxx.ipa to App Store Connect
> app-store-connect publish --path /Users/builder/clone/build/ios/ipa/xxx.ipa

Publish "/Users/builder/clone/build/ios/ipa/xxx.ipa" to App Store Connect
App name: xxx
Bundle identifier: com.app.xxx-staging
Certificate expires: 2024-05-12T00:33:02.000+0000
Distribution type: App Store
Min os version: 12.0
Provisioned devices: N/A
Provisions all devices: No
Supported platforms: iPhoneOS
Version code: 1
Version: 1.0

Upload "/Users/builder/clone/build/ios/ipa/xxx.ipa" to App Store Connect
{
    "tool-version": "6.315.14315",
    "tool-path": "/Applications/Xcode-14.3.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/Frameworks/AppStoreService.framework",
    "os-version": "13.3.1",
    "product-errors": [
        {
            "message": "The provided entity includes an attribute with a value that has already been used",
            "userInfo": {
                "NSUnderlyingError": "Error Domain=IrisAPI Code=-19241 \"The provided entity includes an attribute with a value that has already been used\" UserInfo={status=409, detail=The bundle version must be higher than the previously uploaded version., source={\n    pointer = \"/data/attributes/cfBundleVersion\";\n}, id=1111585a-67a3-4d1a-abec-e9858b6045b0, code=ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE, title=The provided entity includes an attribute with a value that has already been used, meta={\n    previousBundleVersion = 16;\n}, NSLocalizedDescription=The provided entity includes an attribute with a value that has already been used, NSLocalizedFailureReason=The bundle version must be higher than the previously uploaded version.}",
                "NSLocalizedDescription": "The provided entity includes an attribute with a value that has already been used",
                "previousBundleVersion": "16",
                "iris-code": "ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE",
                "NSLocalizedFailureReason": "The bundle version must be higher than the previously uploaded version: \u201816\u2019. (ID: 1111585a-67a3-4d1a-abec-e9858b6045b0)"
            },
            "code": -19232
        }
    ]
}
Failed to upload archive at "/Users/builder/clone/build/ios/ipa/xxx.ipa"
Failed to publish /Users/builder/clone/build/ios/ipa/xxx.ipa

Failed to publish xxx.ipa to App Store Connect.

Build failed :|


Publishing failed :|
Failed to publish xxx.ipa to App Store Connect.
```

**Comment by devsideal** [comment]
Hi All, 

What is the solution for this error?
I'm also getting same error, I checked with setting the hard coded versions also like:

```
      - name: Flutter build ipa
        script: |
            flutter build ipa --release \
              --build-name=1.6.4 \
              --build-number=2 \
              --export-options-plist=/Users/builder/export_options.plist
```

```
{
    "tool-version": "6.315.14315",
    "tool-path": "/Applications/Xcode-14.3.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/Frameworks/AppStoreService.framework",
    "os-version": "13.3.1",
    "product-errors": [
        {
            "message": "The provided entity includes an attribute with a value that has already been used",
            "userInfo": {
                "NSUnderlyingError": "Error Domain=IrisAPI Code=-19241 \"The provided entity includes an attribute with a value that has already been used\" UserInfo={status=409, detail=The bundle version must be higher than the previously uploaded version., source={\n    pointer = \"/data/attributes/cfBundleVersion\";\n}, id=ab65abfe-9e8d-4dac-b6e9-0c36a77b86a1, code=ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE, title=The provided entity includes an attribute with a value that has already been used, meta={\n    previousBundleVersion = 1;\n}, NSLocalizedDescription=The provided entity includes an attribute with a value that has already been used, NSLocalizedFailureReason=The bundle version must be higher than the previously uploaded version.}",
                "NSLocalizedDescription": "The provided entity includes an attribute with a value that has already been used",
                "previousBundleVersion": "1",
                "iris-code": "ENTITY_ERROR.ATTRIBUTE.INVALID.DUPLICATE",
                "NSLocalizedFailureReason": "The bundle version must be higher than the previously uploaded version: \u20181\u2019. (ID: ab65abfe-9e8d-4dac-b6e9-0c36a77b86a1)"
            },
            "code": -19232
        }
    ]
}
Failed to upload archive at "/Users/builder/clone/build/ios/ipa/xxxx.ipa"
Failed to publish /Users/builder/clone/build/ios/ipa/xxxx.ipa

Failed to publish xxxx.ipa to App Store Connect.

Build failed :|


Publishing failed :|
Failed to publish xxxx.ipa to App Store Connect.
```

**Comment by appsowner** [comment]
Hello,

This solution was work for me .

  - name: Flutter build ipa and automatic versioning
        script: |
            flutter build ipa --release \
            --build-name=1.0.$PROJECT_BUILD_NUMBER \
            --build-number=$PROJECT_BUILD_NUMBER \
            --export-options-plist=/Users/builder/export_options.plist

Regards
